google/cloud/forseti/notifier/notifiers/email_violations.py

```diff
@@ -121,11 +129,21 @@ def _compose(self, **kwargs):
 
         email_map = {}
 
-        attachment = self._make_attachment()
-        subject, content = self._make_content()
-        email_map['subject'] = subject
-        email_map['content'] = content
-        email_map['attachment'] = attachment
+        data_format = self.notification_config.get('data_format', 'csv')
+        if data_format not in ['json', 'csv']:
+            LOGGER.error(
```

Comment:
It would be nicer to raise a custom exception here.  Perhaps something like `UnsupportedNotificationDataFormat`, and make the caller handle this via try/except.

Then, you can remove the `else` here.

Reply:
Done.